source/h2_decoder.c

```diff
@@ -1534,6 +1534,9 @@ static struct aws_h2err s_state_fn_connection_preface_string(
 }
 
 void aws_h2_decoder_set_setting_header_table_size(struct aws_h2_decoder *decoder, uint32_t data) {
+    /* Set the max_dynamic_table_size for hpack, but we will not update the dynamic table size until we receive dynamic
+     * table size update from header block */
+    aws_hpack_set_max_table_size(decoder->hpack, data, false /*udpate table size*/);
     decoder->settings.header_table_size = data;
```

Comment:
seems like we can just remove the `decoder->settings.header_table_size` variable? It serves no purpose